kratos/mpi/includes/mpi_data_communicator.h

```diff
@@ -209,6 +209,16 @@ class KRATOS_API(KRATOS_MPI_CORE) MPIDataCommunicator: public DataCommunicator
      */
     static MPIDataCommunicator::UniquePointer Create(MPI_Comm MPIComm);
 
+    /**
+     * @brief Initializes the rank list.
+     * @details This function is responsible for initializing the rank list. You should call this function before using some ranking-related operations with sub-DataCommunicator.
+     */
+    void InitializeRanksList() const override;
```

Comment:
```suggestion
    void InitializeRanksList() override;
```

I dont think `Initialize...` methods can/should be const

Reply:
The thing is that the class is usually always const, so the method shpuld be. Anycase, i agree with @matekelemen and add this to the constructor